src/options/editor/file-operations.ts

```diff
@@ -0,0 +1,83 @@
+import { showAlert } from "./alerts";
+
+export const generateDefaultFilename = (): string => {
+  const date = new Date();
+  const timestamp = date.toISOString().replace(/[:.]/g, "-").slice(0, -5);
+  return `blyrics-theme-${timestamp}.css`;
+};
+
+export const saveCSSToFile = (css: string, defaultFilename: string): void => {
+  chrome.permissions.contains({ permissions: ["downloads"] }, hasPermission => {
+    if (hasPermission) {
+      downloadFile(css, defaultFilename);
+    } else {
+      chrome.permissions.request({ permissions: ["downloads"] }, granted => {
+        if (granted) {
+          downloadFile(css, defaultFilename);
+        } else {
+          fallbackSaveMethod(css, defaultFilename);
+        }
+      });
+    }
+  });
+};
+
+const downloadFile = (css: string, defaultFilename: string): void => {
+  const blob = new Blob([css], { type: "text/css" });
+  const url = URL.createObjectURL(blob);
+
+  if (chrome.downloads) {
+    chrome.downloads
+      .download({
+        url: url,
+        filename: defaultFilename,
+        saveAs: true,
+      })
+      .then(() => {
+        showAlert("CSS file save dialog opened. Choose where to save your file.");
+        URL.revokeObjectURL(url);
+      })
+      .catch(error => {
+        console.log(error);
+        showAlert("Error saving file. Please try again.");
+        URL.revokeObjectURL(url);
+      });
+  } else {
+    fallbackSaveMethod(css, defaultFilename);
+  }
+};
+
+const fallbackSaveMethod = (css: string, defaultFilename: string): void => {
+  const blob = new Blob([css], { type: "text/css" });
+  const url = URL.createObjectURL(blob);
+
+  const a = document.createElement("a");
+  a.href = url;
+  a.download = defaultFilename;
+
+  document.body.appendChild(a);
+  a.click();
+  document.body.removeChild(a);
+
+  setTimeout(() => URL.revokeObjectURL(url), 100);
```

Comment:
Feels a little cursed, but makes sense ig

Reply:
yea lmao this ones iffy but we should ideally not hit this anyways, legacy code